pkgs/servers/quagga/default.nix

```diff
@@ -0,0 +1,72 @@
+{ stdenv, fetchurl, libcap, libnl, readline, net_snmp, less, perl, texinfo }:
+
+stdenv.mkDerivation rec {
+  name = "quagga-${version}";
+  version = "0.99.24.1";
+
+  src = fetchurl {
+    url = "mirror://savannah/quagga/${name}.tar.gz";
+    sha256 = "0kvmc810m7ssrvgb3213271rpywyxb646v5bzjl1jl88vx3imbl4";
+  };
+
+  buildInputs =
+    [ readline net_snmp ]
+    ++ stdenv.lib.optionals stdenv.isLinux [ libcap libnl ];
+
+  nativeBuildInputs = [ perl texinfo ];
+
+  configureFlags = [
+    "--sysconfdir=/etc/quagga"
+    "--localstatedir=/run/quagga"
+    "--sbindir=$(out)/libexec/quagga"
+    "--disable-exampledir"
+    "--enable-user=quagga"
+    "--enable-group=quagga"
```

Comment:
The 'quagga' user and group are used by the daemons to drop privileges, they can be overridden with command line arguments; a future NixOS quagga module (wip) will allow the user to override the defaults.

The 'quaggavty' group allows its members to use the 'vtysh' tool to connect to the daemons and is hardcoded at compile-time.